vernac/comFixpoint.mli

```diff
@@ -16,14 +16,29 @@ open Vernacexpr
 
 val do_mutually_recursive
   :  ?pm:Declare.OblState.t
+     (* Obligation mode turns unresolved evars into obligations *)
```

Comment:
Shouldn't these be `**`

Reply:
I don't think that odoc supports `(** *)` comments in between types. If I put `(**` I get this:
<img width="807" alt="odoc-fixpoint" src="https://github.com/coq/coq/assets/460771/3d9b9c39-db82-4c4c-bf8f-72932e7e792e">, that is, it keeps only the last `(**` comment.

Reply:
Hmm I guess the correct way to do this would be in the comment attached to `do_mutually_recrusive`. You could write something like this:
```
(** [do_mutually_recursive ~program_mode ~poly recursives_exp] returns open obligations and open proofs if any. It takes the following arguents:
  - [program_ode] ...
  - [poly] ...
  - [recursives_exp] ...
  - [pm] is an optional argument that turns unresolved evars into obligations.
  - [use_interferene_hook] ...
 *)
val do_mutually_recursive
  :  ?pm:Declare.OblState.t
  -> program_mode:bool
  -> ?use_inference_hook:bool
  -> ?scope:Locality.definition_scope
  -> ?clearbody:bool
  -> poly:bool
  -> ?typing_flags:Declarations.typing_flags
  -> ?user_warns:UserWarn.t
  -> ?using:Vernacexpr.section_subset_expr
  -> recursives_expr
  -> Declare.OblState.t option * Declare.Proof.t option
```
The reason to do it this way is so that you can see how to use the function on hover. For instance, ocaml-lsp shows the odoc description of a function and having the full manual here can help without having to jump to the mli.

Reply:
As an example, here is the documentation of a variant in dune: https://github.com/ocaml/dune/blob/1e8b005dfe7d852887e0eb1d32e0260c9ef70d31/otherlibs/stdune/src/path.mli#L415-L422

Reply:
Somehow, I'm not convinced that it is where documentation is worth. These different functions (see e.g. also `ComDefinition.do_definition`) have the same arguments, 8 of them being actually components of a `Declare.Info.t`.

I feel that it would be better to properly document `Declare.Info.t` and to replace most of the arguments by an `Info.t`.